frontend/src/features/notes/list/NoteListView.tsx

```diff
@@ -31,16 +36,25 @@ export const NoteListView: React.FunctionComponent<INoteListViewProps> = (
   const [sort, setSort] = React.useState<TableSort<Api_Note>>({});
   const [noteResult, setNoteResult] = React.useState<Api_Note[]>([]);
 
-  React.useEffect(() => {
+  const [isAddNotesOpened, openAddNotes, closeAddNotes] = useModalManagement();
+  const [isViewNotesOpened, openViewNotes, closeViewNotes] = useModalManagement();
+
+  const fetchNotes = useCallback(async () => {
     setIsLoading(true);
-    getNotes(type, entityId)
-      .then(({ data }) => {
-        if (data && isMounted()) {
-          setNoteResult(data);
-        }
-      })
-      .finally(() => setIsLoading(false));
-  }, [type, entityId, isMounted, getNotes]);
+    try {
+      const { data } = await getNotes(type, entityId);
+      setNoteResult(data);
+      if (data && isMounted()) {
+        setNoteResult(data);
+      }
+    } finally {
+      setIsLoading(false);
```

Comment:
instead of handling the loading state locally it might be better to use the general hook that performs the fetch and provides a loading indicator

Reply:
yeah - I was just cleaning up a bit the notes list view that Mahesh wrote. We can expand on that later as tech debt I think